int32 comtype
float32 targetdist
